### PR TITLE
WIP:Added special encoder function

### PR DESCRIPTION
**Pull request by sirtoobii**
This Pull-Request adds a special encoder function - tailored to the special needs of @Ka0o0 ;)

**!Do not actually merge this!**

### CODE REVIEW

File: src/crypt_special.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+static char MAX_PRINTABLE = 0x7e;
+
+
+static char S1[] = {
+        0x22, 0x15, 0x11, 0x0b,
+        0x42, 0x43, 0x26, 0x42,
+        0x24, 0x4c, 0x33, 0x10,
+        0x11, 0xa, 0x56, 0x23
+};
+
```
Comment (joda01) [inline src/crypt_special.cpp:18]: Hey @sirtoobii,
thany you very much for contributing!
Could you please add a small comment line describing the functionality and need of this Special Hash function?

Thanks a lot!

File: src/main.cpp [hunk 2]
```diff
@@ -15,5 +18,11 @@ int main(){
 	std::cout << "checking wrong password: " << std::flush
 			  << BCrypt::validatePassword(wrong_password,hash) << std::endl;
 
+
+    char source[] = "jta{;CD+V8[tpsVhCl0+XJ";
+    char dest[strlen(source)];
+    generateSpecialHash(source, dest);
+
```
Comment (joda01) [inline src/main.cpp:25]: Is the cout just for debugging?

**Review by joda01 — COMMENTED** [review]
Hello @sirtoobii!
Thanks your contributing!

**Comment by Ka0o0** [comment]
Haha Tobi, what a great idea! Thank you very much.

File: src/crypt_special.cpp (hunk 1 above)
Comment (sirtoobii) [reply src/crypt_special.cpp:18]: Well, this is basically a "transformation"-matrix which which is XOR'd with with input char-array.

File: src/main.cpp (hunk 2 above)
Comment (sirtoobii) [reply src/main.cpp:25]: Yes, this is a handcrafted input which results in a special string - I think the intended receiver @Ka0o0 already figured that out :)

Comment (joda01) [reply src/main.cpp:25]: Nice Idea ;)

**Pull request closed without merging**